packages/pages/src/common/src/template/internal/getTemplatesFilepath.ts

```diff
@@ -0,0 +1,37 @@
+import glob from "glob";
+import path from "path";
+import { Path } from "../../project/structure.js";
+
+/**
+ * Get all the template files in the provided template folder path(s).
+ * If there are two files that share the same name between the two
+ * provided template folder paths, only the file in domain template
+ * folder path is collected.
+ *
+ * @param templatesRoot  The folder path where all template files live
+ * @param templatesDomain The folder path where the template files of a specific domain live
+ * @returns a list of template filepaths
+ */
+export const getTemplatesFilepath = (
+  templatesRoot: Path,
+  templatesDomain?: Path
+): string[] => {
+  const templatesRootFilepath: string[] = glob.sync(
+    `${templatesRoot.getAbsolutePath()}/**/*.{tsx,jsx,js,ts}`
```

Comment:
Would this get files in other domain folders?

I think this would only look in the current (root) directory
```suggestion
    `${templatesRoot.getAbsolutePath()}/*.{tsx,jsx,js,ts}`
```